examples/experimental/dagster-airlift/dagster_airlift/core/partition_utils.py

```diff
@@ -0,0 +1,53 @@
+from dagster import (
+    AssetKey,
+    PartitionsDefinition,
+    TimeWindowPartitionsDefinition,
+    _check as check,
+)
+from dagster._time import get_timezone
+
+from dagster_airlift.core.airflow_instance import TaskInstance
+
+
+def get_partition_key_from_task_instance(
+    partitions_def: PartitionsDefinition, task_instance: TaskInstance, asset_key: AssetKey
+) -> str:
+    partitions_def = check.inst(
+        partitions_def,
+        TimeWindowPartitionsDefinition,
+        "Only time window-partitioned assets are supported.",
+    )
+    logical_date = task_instance.logical_datetime
```

Comment:
To make this super clear, I think we should prefix local variables that store airflow's concept of date with `airflow_`. It is very easy IMO to get all the different notions of time confused in this function. Let's do a readability pass.